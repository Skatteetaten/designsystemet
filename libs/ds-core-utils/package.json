{
  "name": "@skatteetaten/ds-core-utils",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.1.1",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14"
  },
  "scripts": {}
}
