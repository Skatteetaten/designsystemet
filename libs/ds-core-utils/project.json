{
  "name": "ds-core-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-core-utils/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-core-utils",
        "tsConfig": "libs/ds-core-utils/tsconfig.lib.json",
        "project": "libs/ds-core-utils/package.json",
        "entryFile": "libs/ds-core-utils/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-core-utils/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-core-utils/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-core-utils"],
      "options": {
        "jestConfig": "libs/ds-core-utils/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-core-utils/**/*.css",
          "libs/ds-core-utils/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-core-utils"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-core-utils {args.switches}"
      }
    }
  }
}
