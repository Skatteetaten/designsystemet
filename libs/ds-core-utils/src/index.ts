export * from './SkatteetatenLogo/SkatteetatenLogo';
export * from './base-props.types';
export * from './defaults';
export * from './ExternalLayout/ExternalLayout';
export * from './InternalLayout/InternalLayout';
export * from './i18n';
export * from './Separator/Separator';
export * from './hooks/hooks';
export * from './formatter/formatter';
