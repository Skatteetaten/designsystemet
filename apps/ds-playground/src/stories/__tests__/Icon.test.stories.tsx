import { sizeArr } from '@skatteetaten/ds-core-utils';
import {
  Icon,
  IconComponentCommonProps,
  AccountChildSVGpath,
  AndreForholdSVGpath,
} from '@skatteetaten/ds-icons';
import { Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { SystemSVGPaths } from '../utils/icon.systems';
import { ThemeSVGPaths } from '../utils/icon.themes';

const meta = {
  component: Icon,
  title: 'Tester/Icon',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    size: {
      table: { disable: true },
      control: 'radio',
      options: [...sizeArr].slice(1),
    },
    svgPath: {
      table: { disable: true },
      options: Object.keys(SystemSVGPaths),
      mapping: SystemSVGPaths,
    },
    title: { table: { disable: true } },
    variant: {
      table: { disable: true },
      control: 'text',
    },
    // Aria
    ariaLabel: { table: { disable: true } },
  },
} satisfies Meta<typeof Icon>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs: IconComponentCommonProps = {
  svgPath: AccountChildSVGpath,
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: SVGSVGElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes(FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon).toHaveAttribute('id', 'htmlid');
    await expect(icon).toHaveClass('dummyClassname');
    await expect(icon).toHaveAttribute('lang', 'nb');
    await expect(icon).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Default - Variant SystemIcon (A1, B1, B5)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    svgPath: {
      table: { disable: false },
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon).toBeInTheDocument();
    await expect(icon).toHaveAttribute('viewBox', '0 0 24 24');
    await expect(icon).toHaveAttribute('aria-hidden', 'true');
    await expect(icon).not.toHaveAttribute('aria-label');
    await expect(icon).not.toHaveAttribute('aria-labelledby');
  },
} satisfies Story;

export const WithVariant = {
  name: 'With Variant ThemeIcon (A1)',
  args: {
    ...defaultArgs,
    svgPath: AndreForholdSVGpath,
    variant: 'themeIcon',
  },
  argTypes: {
    svgPath: {
      table: { disable: false },
      options: Object.keys(ThemeSVGPaths),
      mapping: ThemeSVGPaths,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon).toBeInTheDocument();
    await expect(icon).toHaveAttribute('viewBox', '0 0 48 48');
    await expect(icon.classList.toString()).toContain('themeIcon_medium');
  },
} satisfies Story;

export const WithCustomSVG = {
  name: 'With Custom SVG (A4)',
  args: {
    ...defaultArgs,
    svgPath: <path d={'M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2Z'} />,
  },
  argTypes: {
    svgPath: {
      table: { disable: false },
      control: { disable: true },
    },
  },
} satisfies Story;

export const WithTitle = {
  name: 'With Title (B3)',
  args: {
    ...defaultArgs,
    title: 'Min custom title beskrivelse',
  },
  argTypes: {
    title: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img');
    await expect(icon).toBeInTheDocument();
    await expect(icon).toHaveAttribute('aria-hidden', 'false');
    await expect(icon).toHaveAttribute('aria-labelledby');
    await expect(icon).not.toHaveAttribute('aria-label');

    const titleText = 'Min custom title beskrivelse';
    const titleNode = canvas.getByTitle(titleText);
    await expect(titleNode).toBeInTheDocument();
    await expect(titleNode).toHaveAttribute('id');
    const svgNode = canvas.getByLabelText(titleText, { selector: 'svg' });
    await expect(svgNode).toBeInTheDocument();
  },
} satisfies Story;

export const WithAriaLabel = {
  name: 'With AriaLabel (B4)',
  args: {
    ...defaultArgs,
    ariaLabel: 'min custom aria-label beskrivelse',
  },
  argTypes: {
    ariaLabel: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img');
    await expect(icon).toBeInTheDocument();
    await expect(icon).toHaveAttribute('aria-hidden', 'false');
    await expect(icon).toHaveAttribute(
      'aria-label',
      'min custom aria-label beskrivelse'
    );
    await expect(icon).not.toHaveAttribute('aria-labelledby');

    const title = icon.querySelector('title');
    await expect(title).not.toBeInTheDocument();
  },
} satisfies Story;

export const SystemIconSizeSmall = {
  name: 'With Size Small - SystemIcon (A2)',
  args: {
    ...defaultArgs,
    svgPath: AccountChildSVGpath,
    variant: 'systemIcon',
    size: 'small',
  },
  argTypes: {
    size: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon.classList.toString()).toContain('systemIcon_small');
  },
} satisfies Story;

export const SystemIconSizeMedium = {
  name: 'With Size Medium - SystemIcon (A2)',
  args: {
    ...defaultArgs,
    svgPath: AccountChildSVGpath,
    variant: 'systemIcon',
    size: 'medium',
  },
  argTypes: {
    size: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon.classList.toString()).toContain('systemIcon_medium');
  },
} satisfies Story;

export const SystemIconSizeLarge = {
  name: 'With Size Large - SystemIcon (A2)',
  args: {
    ...defaultArgs,
    svgPath: AccountChildSVGpath,
    variant: 'systemIcon',
    size: 'large',
  },
  argTypes: {
    size: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon.classList.toString()).toContain('systemIcon_large');
  },
} satisfies Story;

export const SystemIconSizeExtraLarge = {
  name: 'With Size Extra Large - SystemIcon (A2)',
  args: {
    ...defaultArgs,
    svgPath: AccountChildSVGpath,
    variant: 'systemIcon',
    size: 'extraLarge',
  },
  argTypes: {
    size: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon.classList.toString()).toContain('systemIcon_extraLarge');
  },
} satisfies Story;

export const ThemeIconSizeMedium = {
  name: 'With Size Medium - ThemeIcon (A3)',
  args: {
    ...defaultArgs,
    svgPath: AndreForholdSVGpath,
    variant: 'themeIcon',
    size: 'medium',
  },
  argTypes: {
    size: {
      table: { disable: false },
      options: [sizeArr[2], sizeArr[3]],
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon.classList.toString()).toContain('themeIcon_medium');
  },
} satisfies Story;

export const ThemeIconSizeLarge = {
  name: 'With Size Large - ThemeIcon (A3)',
  args: {
    ...defaultArgs,
    svgPath: AndreForholdSVGpath,
    variant: 'themeIcon',
    size: 'large',
  },
  argTypes: {
    size: {
      table: { disable: false },
      options: [sizeArr[2], sizeArr[3]],
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon.classList.toString()).toContain('themeIcon_large');
  },
} satisfies Story;
