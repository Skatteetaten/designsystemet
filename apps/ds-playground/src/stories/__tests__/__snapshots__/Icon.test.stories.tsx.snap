// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Icon Defaults play-test 1`] = `
<div data-test-block="true">
  <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10"
       viewbox="0 0 24 24"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M12.5 11a2 2 0 1 0-.09 3.998A2 2 0 0 0 12.5 11Zm0-2c1.63 0 3.12.35 4.5 1.05 1.34.71 2 1.56 2 2.56v5.77c0 1.12-.86 2.06-2.61 2.81V19c0-.95-.86-1.62-2.58-2.03-.56-.13-1-.19-1.31-.19-.87 0-1.7.17-2.46.52-.77.34-1.23.78-1.38 1.31 1.34.53 2.62.8 3.84.8l1-.1v2.63l-1 .06a9.68 9.68 0 0 1-3.89-.81C6.86 20.44 6 19.5 6 18.38v-5.77c0-1 .66-1.85 2-2.56C9.38 9.35 10.88 9 12.5 9Zm0-7a3 3 0 1 1 0 6 3 3 0 0 1 0-6Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon SystemIconSizeExtraLarge play-test 1`] = `
<div data-test-block="true">
  <svg class="_systemIcon_dry7n_2 _systemIcon_extraLarge_dry7n_20"
       viewbox="0 0 24 24"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M12.5 11a2 2 0 1 0-.09 3.998A2 2 0 0 0 12.5 11Zm0-2c1.63 0 3.12.35 4.5 1.05 1.34.71 2 1.56 2 2.56v5.77c0 1.12-.86 2.06-2.61 2.81V19c0-.95-.86-1.62-2.58-2.03-.56-.13-1-.19-1.31-.19-.87 0-1.7.17-2.46.52-.77.34-1.23.78-1.38 1.31 1.34.53 2.62.8 3.84.8l1-.1v2.63l-1 .06a9.68 9.68 0 0 1-3.89-.81C6.86 20.44 6 19.5 6 18.38v-5.77c0-1 .66-1.85 2-2.56C9.38 9.35 10.88 9 12.5 9Zm0-7a3 3 0 1 1 0 6 3 3 0 0 1 0-6Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon SystemIconSizeLarge play-test 1`] = `
<div data-test-block="true">
  <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15"
       viewbox="0 0 24 24"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M12.5 11a2 2 0 1 0-.09 3.998A2 2 0 0 0 12.5 11Zm0-2c1.63 0 3.12.35 4.5 1.05 1.34.71 2 1.56 2 2.56v5.77c0 1.12-.86 2.06-2.61 2.81V19c0-.95-.86-1.62-2.58-2.03-.56-.13-1-.19-1.31-.19-.87 0-1.7.17-2.46.52-.77.34-1.23.78-1.38 1.31 1.34.53 2.62.8 3.84.8l1-.1v2.63l-1 .06a9.68 9.68 0 0 1-3.89-.81C6.86 20.44 6 19.5 6 18.38v-5.77c0-1 .66-1.85 2-2.56C9.38 9.35 10.88 9 12.5 9Zm0-7a3 3 0 1 1 0 6 3 3 0 0 1 0-6Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon SystemIconSizeMedium play-test 1`] = `
<div data-test-block="true">
  <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10"
       viewbox="0 0 24 24"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M12.5 11a2 2 0 1 0-.09 3.998A2 2 0 0 0 12.5 11Zm0-2c1.63 0 3.12.35 4.5 1.05 1.34.71 2 1.56 2 2.56v5.77c0 1.12-.86 2.06-2.61 2.81V19c0-.95-.86-1.62-2.58-2.03-.56-.13-1-.19-1.31-.19-.87 0-1.7.17-2.46.52-.77.34-1.23.78-1.38 1.31 1.34.53 2.62.8 3.84.8l1-.1v2.63l-1 .06a9.68 9.68 0 0 1-3.89-.81C6.86 20.44 6 19.5 6 18.38v-5.77c0-1 .66-1.85 2-2.56C9.38 9.35 10.88 9 12.5 9Zm0-7a3 3 0 1 1 0 6 3 3 0 0 1 0-6Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon SystemIconSizeSmall play-test 1`] = `
<div data-test-block="true">
  <svg class="_systemIcon_dry7n_2 _systemIcon_small_dry7n_5"
       viewbox="0 0 24 24"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M12.5 11a2 2 0 1 0-.09 3.998A2 2 0 0 0 12.5 11Zm0-2c1.63 0 3.12.35 4.5 1.05 1.34.71 2 1.56 2 2.56v5.77c0 1.12-.86 2.06-2.61 2.81V19c0-.95-.86-1.62-2.58-2.03-.56-.13-1-.19-1.31-.19-.87 0-1.7.17-2.46.52-.77.34-1.23.78-1.38 1.31 1.34.53 2.62.8 3.84.8l1-.1v2.63l-1 .06a9.68 9.68 0 0 1-3.89-.81C6.86 20.44 6 19.5 6 18.38v-5.77c0-1 .66-1.85 2-2.56C9.38 9.35 10.88 9 12.5 9Zm0-7a3 3 0 1 1 0 6 3 3 0 0 1 0-6Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon ThemeIconSizeLarge play-test 1`] = `
<div data-test-block="true">
  <svg class="_themeIcon_dry7n_26 _themeIcon_large_dry7n_38"
       viewbox="0 0 48 48"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M34.519 8.316c.908 0 1.62.75 1.62 1.66V35.93a1.67 1.67 0 0 1-1.66 1.66h-.79v.79c0 .868-.672 1.58-1.501 1.58h-18.41c-.83 0-1.501-.712-1.501-1.58V12.266c0-.87.671-1.58 1.5-1.58h1.186v-.712c0-.908.75-1.659 1.66-1.659l17.896.001Zm-11.062 4.82h-8.692v24.296H31.24V21.274h-7.783v-8.138ZM19.862 25.5v3.042H16.82V25.5h3.042Zm4.661 0v3.042h-3.042V25.5h3.042Zm4.662 0v3.042h-3.042V25.5h3.042Zm-3.279-11.536v4.86h4.267l-4.267-4.86Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon ThemeIconSizeMedium play-test 1`] = `
<div data-test-block="true">
  <svg class="_themeIcon_dry7n_26 _themeIcon_medium_dry7n_33"
       viewbox="0 0 48 48"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M34.519 8.316c.908 0 1.62.75 1.62 1.66V35.93a1.67 1.67 0 0 1-1.66 1.66h-.79v.79c0 .868-.672 1.58-1.501 1.58h-18.41c-.83 0-1.501-.712-1.501-1.58V12.266c0-.87.671-1.58 1.5-1.58h1.186v-.712c0-.908.75-1.659 1.66-1.659l17.896.001Zm-11.062 4.82h-8.692v24.296H31.24V21.274h-7.783v-8.138ZM19.862 25.5v3.042H16.82V25.5h3.042Zm4.661 0v3.042h-3.042V25.5h3.042Zm4.662 0v3.042h-3.042V25.5h3.042Zm-3.279-11.536v4.86h4.267l-4.267-4.86Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon WithAriaLabel play-test 1`] = `
<div data-test-block="true">
  <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10"
       viewbox="0 0 24 24"
       focusable="false"
       aria-label="min custom aria-label beskrivelse"
       aria-hidden="false"
       role="img"
  >
    <path d="M12.5 11a2 2 0 1 0-.09 3.998A2 2 0 0 0 12.5 11Zm0-2c1.63 0 3.12.35 4.5 1.05 1.34.71 2 1.56 2 2.56v5.77c0 1.12-.86 2.06-2.61 2.81V19c0-.95-.86-1.62-2.58-2.03-.56-.13-1-.19-1.31-.19-.87 0-1.7.17-2.46.52-.77.34-1.23.78-1.38 1.31 1.34.53 2.62.8 3.84.8l1-.1v2.63l-1 .06a9.68 9.68 0 0 1-3.89-.81C6.86 20.44 6 19.5 6 18.38v-5.77c0-1 .66-1.85 2-2.56C9.38 9.35 10.88 9 12.5 9Zm0-7a3 3 0 1 1 0 6 3 3 0 0 1 0-6Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon WithAttributes play-test 1`] = `
<div data-test-block="true">
  <svg id="htmlid"
       class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 dummyClassname"
       lang="nb"
       data-testid="123ID"
       viewbox="0 0 24 24"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M12.5 11a2 2 0 1 0-.09 3.998A2 2 0 0 0 12.5 11Zm0-2c1.63 0 3.12.35 4.5 1.05 1.34.71 2 1.56 2 2.56v5.77c0 1.12-.86 2.06-2.61 2.81V19c0-.95-.86-1.62-2.58-2.03-.56-.13-1-.19-1.31-.19-.87 0-1.7.17-2.46.52-.77.34-1.23.78-1.38 1.31 1.34.53 2.62.8 3.84.8l1-.1v2.63l-1 .06a9.68 9.68 0 0 1-3.89-.81C6.86 20.44 6 19.5 6 18.38v-5.77c0-1 .66-1.85 2-2.56C9.38 9.35 10.88 9 12.5 9Zm0-7a3 3 0 1 1 0 6 3 3 0 0 1 0-6Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon WithCustomSVG smoke-test 1`] = `
<div data-test-block="true">
  <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10"
       viewbox="0 0 24 24"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon WithRef play-test 1`] = `
<div data-test-block="true">
  <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10"
       viewbox="0 0 24 24"
       focusable="false"
       aria-hidden="true"
       role="img"
       id="dummyIdForwardedFromRef"
  >
    <path d="M12.5 11a2 2 0 1 0-.09 3.998A2 2 0 0 0 12.5 11Zm0-2c1.63 0 3.12.35 4.5 1.05 1.34.71 2 1.56 2 2.56v5.77c0 1.12-.86 2.06-2.61 2.81V19c0-.95-.86-1.62-2.58-2.03-.56-.13-1-.19-1.31-.19-.87 0-1.7.17-2.46.52-.77.34-1.23.78-1.38 1.31 1.34.53 2.62.8 3.84.8l1-.1v2.63l-1 .06a9.68 9.68 0 0 1-3.89-.81C6.86 20.44 6 19.5 6 18.38v-5.77c0-1 .66-1.85 2-2.56C9.38 9.35 10.88 9 12.5 9Zm0-7a3 3 0 1 1 0 6 3 3 0 0 1 0-6Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon WithTitle play-test 1`] = `
<div data-test-block="true">
  <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10"
       viewbox="0 0 24 24"
       focusable="false"
       aria-labelledby="svgtitle-:r5:"
       aria-hidden="false"
       role="img"
  >
    <title id="svgtitle-:r5:">
      Min custom title beskrivelse
    </title>
    <path d="M12.5 11a2 2 0 1 0-.09 3.998A2 2 0 0 0 12.5 11Zm0-2c1.63 0 3.12.35 4.5 1.05 1.34.71 2 1.56 2 2.56v5.77c0 1.12-.86 2.06-2.61 2.81V19c0-.95-.86-1.62-2.58-2.03-.56-.13-1-.19-1.31-.19-.87 0-1.7.17-2.46.52-.77.34-1.23.78-1.38 1.31 1.34.53 2.62.8 3.84.8l1-.1v2.63l-1 .06a9.68 9.68 0 0 1-3.89-.81C6.86 20.44 6 19.5 6 18.38v-5.77c0-1 .66-1.85 2-2.56C9.38 9.35 10.88 9 12.5 9Zm0-7a3 3 0 1 1 0 6 3 3 0 0 1 0-6Z">
    </path>
  </svg>
</div>
`;

exports[`Tester/Icon WithVariant play-test 1`] = `
<div data-test-block="true">
  <svg class="_themeIcon_dry7n_26 _themeIcon_medium_dry7n_33"
       viewbox="0 0 48 48"
       focusable="false"
       aria-hidden="true"
       role="img"
  >
    <path d="M34.519 8.316c.908 0 1.62.75 1.62 1.66V35.93a1.67 1.67 0 0 1-1.66 1.66h-.79v.79c0 .868-.672 1.58-1.501 1.58h-18.41c-.83 0-1.501-.712-1.501-1.58V12.266c0-.87.671-1.58 1.5-1.58h1.186v-.712c0-.908.75-1.659 1.66-1.659l17.896.001Zm-11.062 4.82h-8.692v24.296H31.24V21.274h-7.783v-8.138ZM19.862 25.5v3.042H16.82V25.5h3.042Zm4.661 0v3.042h-3.042V25.5h3.042Zm4.662 0v3.042h-3.042V25.5h3.042Zm-3.279-11.536v4.86h4.267l-4.267-4.86Z">
    </path>
  </svg>
</div>
`;
